chemprop/cli/train.py

```diff
@@ -852,6 +852,19 @@ def train_model(
         )
         trainer.fit(model, train_loader, val_loader)
 
+        columns = get_column_names(
+            args.data_path,
+            args.smiles_columns,
+            args.reaction_columns,
+            args.target_columns,
+            args.ignore_columns,
+            args.splits_column,
+            args.weight_column,
+            args.no_header_row,
+        )
+        input_cols = (args.smiles_columns or []) + (args.reaction_columns or [])
+        target_cols = columns[1:] if len(input_cols) == 0 else columns[len(input_cols) :]
```

Comment:
Can we overwrite the values of `target_cols` in `args` with this? This way, we don't need to pass `columns` and `target_cols` into `evaluate_and_save_predictions`.

Reply:
Perhaps we should do this in process_common_args instead if we want to modify args

Reply:
hmmm also not really, even if we have `args.target_columns = target_cols`, we still need `columns` to save the predictions in `evaluate_and_save_predictions`.

Reply:
`columns` is defined as `input_cols + target_cols`. The `input_cols` can be replaced with `input_cols = (args.smiles_columns or []) + (args.reaction_columns or [])`. So, all the information is in the args.

Reply:
We can't do this in common.py because the file that has the column headers, `args.data_path` belongs to train. It should stay a training argument because you don't need it for predicting. 

I also recommend not overwriting `args.target_columns`, etc. because they are collected together in `format_kwargs` and passed to the data building function `build_splits` which has the same logic to get the column names. 

Instead, I see that we only need the updated `target_cols` two places, in `evaluate_and_save_predictions` and to save the model. I would generate `target_cols` (and by extension `columns`) in both places for now. It duplicates code, but we can go back in another PR and try to fix the code duplication. #882 also needs the column names, so how we get the column names could probably be improved.

Reply:
Yes, it cannot be used in common.py. I am thinking of calling `get_column_names` and overwriting `args.target_columns` right after the data building function `build_splits`. By doing so, `get_column_names` would only need to be called once, instead of being called for each `train_model` call. I am fine with generating `target_cols` in both places for now and improving it in the future.

Reply:
I did the overwriting in process_train_args to follow your suggestion of only calling get_column_names once. I think it is safe to overwrite target columns at the very beginning because `parse_csv` in `cli/utils/parsing.py` does the same thing if target_cols is None. Now that code in parse_csv usually won't be called because we do it in process_train_args.